File: src/main/java/github/zmilla93/core/data/PriceThresholdData.java
package github.zmilla93.core.data;

import github.zmilla93.core.audio.SoundComponent;
import github.zmilla93.core.enums.CurrencyType;

public class PriceThresholdData implements Comparable<PriceThresholdData> {

    public final CurrencyType currencyType;
    public final int quantity;
    public final SoundComponent soundComponent;

    public PriceThresholdData(CurrencyType currencyType, int quantity, SoundComponent soundComponent) {
        this.currencyType = currencyType;
        this.quantity = quantity;
        this.soundComponent = soundComponent;
    }

    @Override
    public int compareTo(PriceThresholdData o) {
        return quantity - o.quantity;
    }
}

File: src/main/java/github/zmilla93/core/data/PriceThresholdData.kt
package github.zmilla93.core.data

import github.zmilla93.core.audio.SoundComponent
import github.zmilla93.core.enums.CurrencyType

class PriceThresholdData(
    @JvmField val currencyType: CurrencyType?,
    @JvmField val quantity: Int,
    @JvmField val soundComponent: SoundComponent
) : Comparable<PriceThresholdData> {

    /** GSON Constructor  */
    constructor() : this(null, 0, SoundComponent())

    override fun compareTo(other: PriceThresholdData): Int {
        return quantity - other.quantity
    }

}

